addons/portal/controllers/thread.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.tools import html2plaintext
+from odoo.http import request
+from odoo.addons.mail.controllers import thread
+
+
+class ThreadController(thread.ThreadController):
+
+    @http.route()
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
+        message = super().mail_message_post(thread_model, thread_id, post_data, context, **kwargs)
+        message_data = message.get("mail.message")
+        if (message_data := message_data and message_data[0]) and request.env[
+            "res.partner"
+        ]._get_portal_partner_from_context():
+            message = {
+                "default_message": html2plaintext(message_data.get("body")),
+                "default_message_id": message_data.get("id"),
+                "default_attachment_ids": message.get("ir.attachment"),
+            }
+        return message
+
+    def _prepare_post_data(self, post_data, thread, special_mentions, **kwargs):
+        post_data = super()._prepare_post_data(post_data, thread, special_mentions, **kwargs)
+        if portal_partner := request.env["res.partner"]._get_portal_partner_from_context():
+            post_data.pop("parent_id", None)
```

Comment:
`parent_id` is only allowed for channel, we should make that check in mail/discuss and then we can remove this line from here